sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileAsyncClient.java

```diff
@@ -427,11 +428,13 @@ private boolean checkDoesNotExistStatusCode(Throwable t) {
             .setIgnoreReadOnly(ignoreReadOnly)
             .setSetArchiveAttribute(setArchiveAttribute);
 
+        final String copySource = Utility.encodeUrlPath(sourceUrl);
```

Comment:
if copy source is passed as a parameter then shouldn't we just encode the whole thing ? I'm wondering why just path.

Reply:
The problem with that is it will become an invalid URI since it will encode the :// in the protocol part and everything. :(

Reply:
Gotcha.